app/src/main/java/com/example/cocktaildb/screen/myrecipe/MyRecipeFragment.kt

```diff
@@ -98,7 +154,75 @@ class MyRecipeFragment : BaseFragment<FragmentMyRecipeBinding>(), MyRecipeContra
         Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
     }
 
-    companion object {
-        fun newInstance() = MyRecipeFragment()
+    private fun navigateToRecipeDetail(recipe: Recipe) {
+        try {
+            val bundle = Bundle().apply {
+                putString("cocktail_id", recipe.id)
+                putString("cocktail_name", recipe.name)
+                putString("cocktail_category", recipe.category)
+                putString("cocktail_alcoholic", recipe.alcoholic.ifEmpty { "Unknown" })
+                putString("cocktail_glass", "Cocktail Glass")
+                putString("cocktail_instructions", recipe.instructions)
+
+                val imageUrl = recipeAdapter.getRecipeImageUrl(recipe.id) ?: ""
+                putString("cocktail_image", imageUrl)
+
+                putStringArray("cocktail_ingredients", emptyArray())
+                putStringArray("cocktail_measures", emptyArray())
+            }
+
+            navigateToDetailFragment(bundle)
+            
+        } catch (e: Exception) {
+            Log.e(TAG, "Error navigating to detail: ${e.message}")
+            Toast.makeText(context, "Error opening recipe details", Toast.LENGTH_SHORT).show()

```

Comment:
```suggestion
            Toast.makeText(context, "Error opening recipe details", Toast.LENGTH_SHORT).show()
```
hard code, dùng string resource